tools/mmp/aot.cs

```diff
@@ -208,7 +208,8 @@ public void Compile (IFileEnumerator files)
 
 			List<string> filesToAOT = GetFilesToAOT (files);
 			Parallel.ForEach (filesToAOT, ParallelOptions, file => {
-				string command = String.Format ("--aot{0} {1}{2}", options.IsHybridAOT ? "=hybrid" : "", IsModern ? "--runtime=mobile " : "", StringUtils.Quote (file));
+				string command = String.Format ("--aot{0} {1}{2}{3}", options.IsHybridAOT ? "=hybrid" : "", 
```

Comment:
This is hard to read. Can we do something like this instead?

```csharp
var command = new StringBuilder ();
command.Append ("--aot");
if (options.IsHybridAOT)
    command.Append ("=hybrid");
command.Append (" ");
if (IsModern)
    command.Append ("--runtime=mobile ");
if (!IsRelease)
    command.Append ("--debug ");
command.Append (StringUtils.Quote (file));
```

I find this much easier to reason with, and additionally diffs are simpler as well.

That said, this is what mtouch does:

1. Always pass `--debug`: https://github.com/xamarin/xamarin-macios/blob/ed57182937ab610850c5e8fc7c2216a3d0346284/tools/mtouch/mtouch.cs#L397
2. There's an mtouch option to enable debugging support per assembly (`--aot=soft-debug`, this is so that the soft debugger works): https://github.com/xamarin/xamarin-macios/blob/ed57182937ab610850c5e8fc7c2216a3d0346284/tools/mtouch/mtouch.cs#L429
3. We pass `--aot=dwarfdebug` so that the AOT compiler emits debug information in dwarf format (which are extracted into the .dSYM) , which makes apps have source code information even in release builds without pdbs:  https://github.com/xamarin/xamarin-macios/blob/ed57182937ab610850c5e8fc7c2216a3d0346284/tools/mtouch/mtouch.cs#L431

I'm not sure if any/all of this is applicable for non-full-aot scenarios though.